Makefile

```diff
@@ -643,6 +643,11 @@ check-kubernetes-yaml: ## Check that the generated Kubernetes YAML configs are u
 check-events: ## Check that events satisfy patterns set in https://github.com/timberio/vector/blob/master/rfcs/2020-03-17-2064-event-driven-observability.md
 	${MAYBE_ENVIRONMENT_EXEC} ./scripts/check-events.sh
 
+##@ Rustdoc
+build-rustdoc: ## Build a subset of Vector's Rustdocs (as specified by the "docs" feature)
+	# This command is mostly intended for use by the build process in timberio/vector-rustdoc
+	${MAYBE_ENVIRONMENT_EXEC} cargo doc --no-deps --no-default-features --features docs
```

Comment:
Is there a specific reason we scope this from the default features?

Reply:
Basically, if you build with plain old `cargo doc`, you end up with *hundreds* of external dependencies listed under the **Crates** section in the left-hand sidebar and a substantially longer build time. Restricting the build to a limited subset of features provides for a less noisy navigation experience. I'm totally open to suggestion on *which* features to include under `docs` and expect that to change over time, but the default "build the universe" approach seems sub-optimal to me.

Reply:
Hmm, I see. Did you maybe run it once without `--no-deps`? Because I don't see the dependencies documented if I just do `cargo doc --no-deps --open` and it completes fairly quickly. Maybe `rm -r target/doc` and rebuild?

Reply:
<img width="1080" alt="Screen Shot 2021-02-25 at 3 27 12 PM" src="https://user-images.githubusercontent.com/316880/109206208-f5007f80-777d-11eb-89d4-e5f39cfa16cd.png">

I just see the `vector` crate.